### PR TITLE
#239 general enhancements other code enhancements small fixes

### LINKED ISSUE #239: General enhancements - Other code enhancements / small fixes
As a developer, I want to have consistent, clear, efficient, and comprehensive code that follows a structure.

To Be Discussed Again:
- [x] SQL Queries should be in the module where they are used.
- [x] Fix how the recommendation calculation is determined in RecommendationAssessment.java. The threshold value set for a label should be the upper value of that threshold. (Make sure the function handles null values).
- [x] Mouse event in generateThumbnails() in SystemsController.java. Create change scene method in UI Utilities.
- [x] Have DecimalFormat used only once instead of multiple times in iniData() in SystemTypeInfoController.java (Use Object in TextConstants)

### PR DESCRIPTION
**Pull request by PaulMicu**
multiple fixes as described in the original issue #239 
 - queries moved 
 - recommendation determination rewrote
 - mouse event adjusted
 - Decimal Format optimized

resolves #239

### CODE REVIEW

File: JavaFXUserInterface/src/main/java/Controllers/SystemTypeInfoController.java [hunk 1]
```diff
@@ -3,22 +3,20 @@
...
-import local.AssetTypeDAOImpl;
 
 import java.net.URL;
-import java.text.DecimalFormat;
 import java.util.Optional;
 import java.util.ResourceBundle;
 
 public class SystemTypeInfoController implements Initializable {
-    private final String SYSTEM_TYPE_LIST = "/Utilities.SystemTypeList";
+    private final String SYSTEM_TYPE_LIST = "/SystemTypeList";
```
Comment (rdsaliba) [inline JavaFXUserInterface/src/main/java/Controllers/SystemTypeInfoController.java:19]: This is the second time it is declared. Would be better to move it in the constants file and call it from there.

File: JavaFXUserInterface/src/main/java/Controllers/AddSystemTypeController.java [hunk 2]
```diff
@@ -4,20 +4,20 @@
...
 import javafx.scene.control.TextFormatter;
-import local.AssetTypeDAOImpl;
 
 import java.net.URL;
 import java.util.ArrayList;
 import java.util.ResourceBundle;
 
 public class AddSystemTypeController implements Initializable {
-    private final String SYSTEM_TYPE_LIST = "/Utilities.SystemTypeList";
+    private final String SYSTEM_TYPE_LIST = "/SystemTypeList";
```
Comment (rdsaliba) [inline JavaFXUserInterface/src/main/java/Controllers/AddSystemTypeController.java:20]: First occurence. Should be declared in the constants file

File: JavaFXUserInterface/src/main/java/Controllers/SystemsController.java [hunk 3]
```diff
@@ -107,7 +99,7 @@ public void attachEvents() {
         systemMenuBtn.setOnMouseClicked(mouseEvent -> uiUtilities.changeScene(mouseEvent, "/Systems"));
 
         //Attach link to systemTypeMenuBtn to go to Utilities.SystemTypeList.fxml
-        systemTypeMenuBtn.setOnMouseClicked(mouseEvent -> uiUtilities.changeScene(mouseEvent, "/Utilities.SystemTypeList"));
+        systemTypeMenuBtn.setOnMouseClicked(mouseEvent -> uiUtilities.changeScene(mouseEvent, "/SystemTypeList"));
```
Comment (rdsaliba) [inline JavaFXUserInterface/src/main/java/Controllers/SystemsController.java:102]: As per my previous comment, this should be called from its constant variable "/SystemTypeList"

**Comment by PaulMicu** [comment]
@rdsaliba yes it should be in the constant file but its out of scope of this issue. We have an issue specifically for that and I figured it will be addressed there. The only reason I even touched those lines was because some of the previous refactoring modified it and broke it.

**Comment by rdsaliba** [comment]
Hi @PaulMicu ,
I went over your changes (there was a lot). I think this covers everything we had discussed in the meeting. Thanks for splitting the commits per what it is changing, made it way easier to review. Let's make sure my comments will be addressed in the other issue you mentioned. Could you referenced it here just to keep track?

**Review by rdsaliba — APPROVED** [review]

**Comment by PaulMicu** [comment]
> Hi @PaulMicu ,
> I went over your changes (there was a lot). I think this covers everything we had discussed in the meeting. Thanks for splitting the commits per what it is changing, made it way easier to review. Let's make sure my comments will be addressed in the other issue you mentioned. Could you referenced it here just to keep track?

ya, the issue is  #238

File: JavaFXUserInterface/src/main/java/Utilities/UIUtilities.java [hunk 4]
```diff
@@ -92,22 +92,10 @@ public void changeScene(MouseEvent mouseEvent, String fxmlFileName) {
...
-                Stage window = (Stage) ((Node) mouseEvent.getSource()).getScene().getWindow();
-                window.setScene(systemInfo);
-                SystemInfoController controller = loader.getController();
-                controller.initData(asset);
-                window.show();
-            }
-        } catch (IOException e) {
-            e.printStackTrace();
+        if (!row.isEmpty()) {
+            changeScene(mouseEvent, fxmlFileName, asset);
```
Comment (Shirwarsame) [inline JavaFXUserInterface/src/main/java/Utilities/UIUtilities.java:96]: Good idea creating a method for screen changes

**Review by Shirwarsame — APPROVED** [review]
@PaulMicu Excellent work. The code works well. As Roy mentioned it was a smart idea to split each commit to its specific issue.

**Pull request merged**